.github/workflows/trunk.yml

```diff
@@ -351,3 +351,90 @@ jobs:
           PYTHON_EXECUTABLE=python ${CONDA_RUN} bash .ci/scripts/test_model.sh "${MODEL_NAME}" "${BUILD_TOOL}" "${BACKEND}"
           echo "::endgroup::"
         done
+
+  test-huggingface-transformers:
+    name: test-huggingface-transformers
+    uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
+    strategy:
+      matrix:
+        hf_model_repo: [google/gemma-2b]
+      fail-fast: false
+    with:
+      secrets-env: "HF_TOKEN_PERIODIC"
```

Comment:
@huydhn Do you know what the secret is for downloading huggingface checkpoints in the CI?

Reply:
There is none in ET atm.  I think we need to create one (make it read-only)

Reply:
Thank you. Let me know when I can update the secrets-env with it. Right now it's just a placeholder I copied from torchchat

Reply:
We will also use it in the benchmark workflow for huggingface models.

Reply:
The way secret is passed to a Linux job is a bit awkward IMO.  You need 3 steps:

1. Add a secret to the repo `bunnylol oss pytorch/executorch` and go to the Settings tab
2. Add `secrets-env` parameter with the name of the secret from step 1
3. Inside the shell script, you can access the secret via `$SECRET_HF_TOKEN_PERIODIC`, notice the `SECRET_` prefix in the variable name

Reply:
Added a read-only token `HF_TOKEN` for ExecuTorch using my personal account to start testing this PR. @huydhn BTW, is there a better account that I should use instead?

Reply:
@huydhn With the newly added secret ([here](https://www.internalfb.com/intern/opensource/github/repo/1868531010203915/repo_settings/secrets/)), the CI will always stuck at `run_with_env_secrets.py` https://github.com/pytorch/executorch/actions/runs/10817669592/job/30011556482?pr=4723. Any pointer to debug the secret?

Reply:
To sum up our chat, the correct way to pass the token to HF is `huggingface-cli login --token $SECRET_EXECUTORCH_HF_TOKEN` according to their doc.  Otherwise, the command hangs to wait for token from shell